### PR TITLE
MacOS samples

### PR DESCRIPTION
**Pull request by BastianBlokland**
Added basic port of the samples to MacOS running on the MoltenVK library. For window-management its using a thin objective-c layer. (source of native obj-c: https://github.com/BastianBlokland/macwindow-dotnet). Nothing close to production ready but it maybe its a start for someone who wants to make a c# vulkan renderer that runs on MacOS.

### CODE REVIEW

File: Samples/MacOS/01-ClearScreen/01-ClearScreen.csproj [hunk 1]
```diff
@@ -0,0 +1,42 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <OutputType>Exe</OutputType>
+    <TargetFramework>netcoreapp2.1</TargetFramework>
+	<AllowUnsafeBlocks>true</AllowUnsafeBlocks>
```
Comment (discosultan) [inline Samples/MacOS/01-ClearScreen/01-ClearScreen.csproj]: mixed tabs and spaces

File: Samples/MacOS/01-ClearScreen/Program.cs [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+        /// </summary>
+        [STAThread]
+        public static void Main()
+        {
+            using (var window = new MacOSWindow(
+                Assembly.GetExecutingAssembly().GetName().Name,
+                new ClearScreenApp()))
+            {
+                window.Initialize();
+				window.Run();
```
Comment (discosultan) [inline Samples/MacOS/01-ClearScreen/Program.cs]: mixed tabs and spaces

File: Samples/MacOS/02-ColoredTriangle/02-ColoredTriangle.csproj [hunk 3]
```diff
@@ -0,0 +1,64 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <OutputType>Exe</OutputType>
+    <TargetFramework>netcoreapp2.1</TargetFramework>
+	<AllowUnsafeBlocks>true</AllowUnsafeBlocks>
```
Comment (discosultan) [inline Samples/MacOS/02-ColoredTriangle/02-ColoredTriangle.csproj]: mixed tabs and spaces

File: Samples/MacOS/03-TexturedCube/03-TexturedCube.csproj [hunk 4]
```diff
@@ -0,0 +1,65 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <OutputType>Exe</OutputType>
+    <TargetFramework>netcoreapp2.1</TargetFramework>
+	<AllowUnsafeBlocks>true</AllowUnsafeBlocks>
```
Comment (discosultan) [inline Samples/MacOS/03-TexturedCube/03-TexturedCube.csproj]: mixed tabs and spaces

File: Samples/MacOS/04-ComputeParticles/04-ComputeParticles.csproj [hunk 5]
```diff
@@ -0,0 +1,75 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <OutputType>Exe</OutputType>
+    <TargetFramework>netcoreapp2.1</TargetFramework>
+	<AllowUnsafeBlocks>true</AllowUnsafeBlocks>
```
Comment (discosultan) [inline Samples/MacOS/04-ComputeParticles/04-ComputeParticles.csproj]: mixed tabs and spaces

File: Samples/Shared/MacOSWindow.cs [hunk 6]
```diff
@@ -0,0 +1,119 @@
...
+namespace VulkanCore.Samples
+{
+    public class MacOSWindow : IVulkanAppHost
+    {
+        private readonly string _title;
+        private readonly Timer _timer = new Timer();
+        private readonly VulkanApp _app;
+
+        private NativeMacOS.NativeApp _nativeApp;
+		private NativeMacOS.NativeWindow _nativeWindow;
```
Comment (discosultan) [inline Samples/Shared/MacOSWindow.cs]: mixed tabs and spaces

File: Src/VulkanLibrary.cs [hunk 7]
```diff
@@ -39,6 +39,10 @@ public static TDelegate GetStaticProc<TDelegate>(string procName)
                 handle = Kernel32GetProcAddress(_handle, procName);
             }
             else if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux))
+            {
+                handle = LibDLGetProcAddress(_handle, procName);
+            }
+			else if (RuntimeInformation.IsOSPlatform(OSPlatform.OSX))
```
Comment (discosultan) [inline Src/VulkanLibrary.cs]: mixed tabs and spaces

**Comment by BastianBlokland** [comment]
Thanks. Sorry for the tabs, we use tabs at the office and forgot that my editor was set to tabs :sweat_smile: Should be fixed now.

**Pull request merged**